From 7ce9c610201f33688c07cc86347ffb6f14a7e06f Mon Sep 17 00:00:00 2001
From: Kirill Kapranov <kirill.kapranov@compulab.co.il>
Date: Sun, 28 Oct 2018 18:20:27 +0200
Subject: [PATCH 90/96] sb-ucm-imx8: dts: Enable SD card reader

Enable SD card reader over multiplexed mmc1 bus, mutually exclusive with Wi-Fi.

Signed-off-by: Kirill Kapranov <kirill.kapranov@compulab.co.il>
---
 arch/arm64/boot/dts/compulab/Makefile              |  1 +
 .../boot/dts/compulab/sb-ucm-imx8-sd-card.dtsi     | 34 ++++++++++++++++++++++
 .../boot/dts/compulab/sbc-ucm-imx8-sd-card.dts     | 25 ++++++++++++++++
 3 files changed, 60 insertions(+)
 create mode 100644 arch/arm64/boot/dts/compulab/sb-ucm-imx8-sd-card.dtsi
 create mode 100644 arch/arm64/boot/dts/compulab/sbc-ucm-imx8-sd-card.dts

diff --git a/arch/arm64/boot/dts/compulab/Makefile b/arch/arm64/boot/dts/compulab/Makefile
index d0d6104..08f3621 100644
--- a/arch/arm64/boot/dts/compulab/Makefile
+++ b/arch/arm64/boot/dts/compulab/Makefile
@@ -4,6 +4,7 @@ dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += sbc-imx8-lvds-dcss.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += sbc-imx8-lvds-lcdif.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += ucm-imx8.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += sbc-ucm-imx8.dtb
+dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += sbc-ucm-imx8-sd-card.dtb
 dtb-$(CONFIG_ARCH_FSL_IMX8MQ) += sbc-ucm-imx8-lvds-dcss.dtb
 
 always		:= $(dtb-y)
diff --git a/arch/arm64/boot/dts/compulab/sb-ucm-imx8-sd-card.dtsi b/arch/arm64/boot/dts/compulab/sb-ucm-imx8-sd-card.dtsi
new file mode 100644
index 0000000..10c5ba5
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/sb-ucm-imx8-sd-card.dtsi
@@ -0,0 +1,34 @@
+/*
+ * Copyright (C) 2018 CompuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+&usdhc2 {
+	pinctrl-names = "default", "state_100mhz", "state_200mhz";
+	pinctrl-0 = <&pinctrl_usdhc2>, <&pinctrl_usdhc2_gpio>;
+	pinctrl-1 = <&pinctrl_usdhc2_100mhz>, <&pinctrl_usdhc2_gpio>;
+	pinctrl-2 = <&pinctrl_usdhc2_200mhz>, <&pinctrl_usdhc2_gpio>;
+	bus-width = <4>;
+	max-frequency = <25000000>;
+	cd-gpios = <&gpio2 12 GPIO_ACTIVE_LOW>;
+	wp-gpios = <&gpio2 20 GPIO_ACTIVE_HIGH>;
+	vmmc-supply = <&reg_usdhc2_vmmc>;
+	no-1-8-v;
+	status = "okay";
+};
+
+&reg_usdhc2_vmmc {
+	gpio = <&gpio1 13 GPIO_ACTIVE_HIGH>;
+	enable-active-high;
+	status = "okay";
+};
+
diff --git a/arch/arm64/boot/dts/compulab/sbc-ucm-imx8-sd-card.dts b/arch/arm64/boot/dts/compulab/sbc-ucm-imx8-sd-card.dts
new file mode 100644
index 0000000..b8381e2
--- /dev/null
+++ b/arch/arm64/boot/dts/compulab/sbc-ucm-imx8-sd-card.dts
@@ -0,0 +1,25 @@
+/*
+ * Copyright (C) 2018 CompuLab Ltd.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ */
+
+/dts-v1/;
+
+#include "ucm-imx8.dtsi"
+#include "sb-ucm-imx8.dtsi"
+#include "sb-ucm-imx8-sd-card.dtsi"
+#include "sb-ucm-imx8-lvds-lcdif.dtsi"
+
+/ {
+	model = "CompuLab UCM-iMX8 on SBC-UCM-iMX8";
+	compatible = "compulab,sbc-ucm-imx8", "compulab,ucm-imx8", "fsl,imx8mq";
+};
-- 
1.9.1

